utilities/infra.py

```diff
@@ -674,3 +705,26 @@ def get_operator_distribution(client: DynamicClient, dsc_name: str = "default-ds
             raise ValueError("DSC release name not found in {dsc_name}")
 
     raise MissingResourceError(f"DSC {dsc_name} not found")
+
+
+def wait_for_serverless_pods_deletion(resource: Project | Namespace, admin_client: DynamicClient | None) -> None:
+    """
+    Wait for serverless pods deletion.
+
+    Args:
+        resource (Project | Namespace): project or namespace
+        admin_client (DynamicClient): admin client.
+
+    Returns:
+        bool: True if we should wait for namespace deletion else False
+
+    """
+    client = admin_client or get_client()
+    for pod in Pod.get(dyn_client=client, namespace=resource.name):
+        if (
+            pod.exists
+            and pod.instance.metadata.annotations.get(Annotations.KserveIo.DEPLOYMENT_MODE)
+            == KServeDeploymentType.SERVERLESS
+        ):
+            LOGGER.info(f"Waiting for {KServeDeploymentType.SERVERLESS} pod {pod.name} to be deleted")
+            pod.wait_deleted(timeout=Timeout.TIMEOUT_1MIN)
```

Comment:
Instead of sequentially waiting for each of them to be deleted, why not check the list of pods that matches this annotation, till the list becomes empty list?

Reply:
because the other pods are being deleted at the same time as the first pod so once we wait for the 1st one to be deleted, the other ones are either deleted or will be deleted soon after.
saving API calls.